lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -1115,6 +1213,42 @@ SubExprInfo ExprEmitter::visitComb(MuxOp op) {
   return {Conditional, signedness};
 }
 
+SubExprInfo ExprEmitter::visitTypeOp(StructCreateOp op) {
+  StructType stype = op.getType().cast<StructType>();
+  os << "'{";
+  size_t i = 0;
+  llvm::interleaveComma(stype.getElements(), os,
+                        [&](const StructType::FieldInfo &field) {
+                          os << field.name << ": ";
+                          emitSubExpr(op.getOperand(i++), Selection);
+                        });
+  os << '}';
+  return {Unary, IsUnsigned};
+}
+
+SubExprInfo ExprEmitter::visitTypeOp(StructExtractOp op) {
+  emitSubExpr(op.input(), Selection);
+  os << '.' << op.field();
+  return {Selection, IsUnsigned};
+}
+
+SubExprInfo ExprEmitter::visitTypeOp(StructInjectOp op) {
+  StructType stype = op.getType().cast<StructType>();
+  os << "'{";
+  llvm::interleaveComma(stype.getElements(), os,
+                        [&](const StructType::FieldInfo &field) {
+                          os << field.name << ": ";
+                          if (field.name == op.field()) {
+                            emitSubExpr(op.newValue(), Selection);
+                          } else {
+                            emitSubExpr(op.input(), Selection);
```

Comment:
This is emitting `op.input()` mutliple times which can lead to expontential blowup in generated verilog.  Please make sure this is handled in isExpressionUnableToInline like SExtOp